frontend/src/app/main/ui/workspace/sidebar/options/menus/layout_container.cljs

```diff
@@ -335,46 +646,139 @@
          ;;on destroy component
          (select-gap nil))))
 
-    [:div.layout-row
-     [:div.gap.row-title "Gap"]
-     [:div.gap-group
-      [:div.gap-row.tooltip.tooltip-bottom-left
-       {:alt "Column gap"}
-       [:span.icon
-        i/auto-gap]
-       [:> numeric-input* {:no-validate true
-                           :placeholder "--"
-                           :on-focus (fn [event]
-                                       (select-gap :column-gap)
-                                       (reset! gap-selected? :column-gap)
-                                       (dom/select-target event))
-                           :on-change (partial set-gap (= :nowrap wrap-type) :column-gap)
-                           :on-blur (fn [_]
-                                      (select-gap nil)
-                                      (reset! gap-selected? :none))
-                           :value (:column-gap gap-value)
-                           :disabled (and (= :nowrap wrap-type) is-col?)}]]
-
-      [:div.gap-row.tooltip.tooltip-bottom-left
-       {:alt "Row gap"}
-       [:span.icon.rotated
-        i/auto-gap]
-       [:> numeric-input* {:no-validate true
-                           :placeholder "--"
-                           :on-focus (fn [event]
-                                       (select-gap :row-gap)
-                                       (reset! gap-selected? :row-gap)
-                                       (dom/select-target event))
-                           :on-change (partial set-gap (= :nowrap wrap-type) :row-gap)
-                           :on-blur (fn [_]
-                                      (select-gap nil)
-                                      (reset! gap-selected? :none))
-                           :value (:row-gap gap-value)
-                           :disabled (and (= :nowrap wrap-type) (not is-col?))}]]]]))
+    (if new-css-system
+      [:div {:class (stl/css :gap-group)}
+       [:div {:class (stl/css-case :row-gap true
+                                   :disabled (and (= :nowrap wrap-type) (not is-col?)))
+              :title "Row gap"}
+        [:span {:class (stl/css :icon)}
+         i/gap-vertical-refactor]
+        [:> numeric-input* {:className (stl/css :numeric-input true)
+                            :no-validate true
+                            :placeholder "--"
+                            :on-focus (fn [event]
```

Comment:
The same here, you pass callbacks as is, which will make the component rerender on all rerenders. This callbacks should be statically defined with use-fn and passed here.

Reply:
This component should be updated in the hole app